jina/types/document/__init__.py

```diff
@@ -135,6 +137,10 @@ def __init__(self, document: Optional[DocumentSourceType] = None,
                              f'if you are trying to set the content '
                              f'you may use "Document(content=your_content)"') from ex
 
+        if self._document.id is None or self._document.id == '':
+            import random
+            self.id = random.randint(0, 9223372036854775807)
```

Comment:
isn't there some Python built-in Integer.Max or something?

Reply:
This is `2**64/2`, which is what should well fit into 64 bit. Apparently, we could directly use `2**64`. Haven't thought too much about it. I'll spend another thought tomorrow.

Reply:
yes but python must have some default value for this.

Reply:
Nope, python does not. `int` is unbounded in python. Anyhow, `numpy` has: `np.iinfo(np.int64).max`, which is `9223372036854775807`. Since we do `int(value).to_bytes(_digest_size, sys.byteorder, signed=True)` to convert it to bytes, this should also be the right number. If I understand correctly, we could also have negative `id`s, but that feels weird. So I restrict for the `numpy` max.

Reply:
yes I am not saying number is wrong, at least have a CONSTANT well documented rather than a magic number.